### DISCUSSION TITLE
The way of creating Variable

### DISCUSSION THREAD

**Discussion by zhuyubei** [biolab/orange3#6803] [Q&A]
I've been reading the source code of Orange3 lately. I noticed that in the domain class, there are methods like SharedComputeValue or ComputeValue used to define how the domain is computed. Initially, I thought they were the primary means of adding Variable values. However, later on, I gradually realized that when I use third-party libraries, calling certain transforms or predicts directly generates new columns, while I can simply create Variables by using ContinuousVariable along with the column name. I'm wondering if using this approach can handle most scenarios.

Thanks!

**Comment by markotoplak** [comment]
Defining variables without compute values is, in most cases, sufficient.

What you lose are automatic transformations. For example, if you process the training data, and then want to predict unprocessed test data. Without compute values you would not be able to, but if compute values are set properly, then the processing of train data can be automatically applied to the test data. Nice to have, but sometimes too much work for the benefit. Most core widgets in Orange are written with these domain transformations though.

**Reply by zhuyubei** [reply]
Thanks for the reply! 

I find __call__ funtion in XXModel class can be used for predicting test data.  So If I can handle the training and prediction function in a proper way, I can use Table.X, Table.Y to set the output directly ? Could this have a better performance?
